### ISSUE TITLE
Add pixelRatio matchesGoldenFile()

### ISSUE THREAD

**Issue by christian-muertz** [flutter/flutter#58226]
I am not to sure with this one, but wouldn't it make sense to be able to pass a pixelRatio for the output image to `matchesGoldenFile`. Currently the golden file has the size of the surface in logical pixels. If I want to have an image that represents the exact pixels of my phone we would need sth. like `pixelRatio` which can be passed into the function as an optin.

**Comment by christian-muertz** [comment]
@Piinks what do you think?

**Comment by christian-muertz** [comment]
Now I am even more confused:
```dart
/// Implementation of RenderBox.paintBounds
Rect get paintBounds => Offset.zero & size;

/// Implementation of RenderView.paintBounds
Rect get paintBounds => Offset.zero & (size * configuration.devicePixelRatio);
```

This results in a strange behaviour for golden file testing. If i pass a RenderView finder into `matchesGoldenFile` I actually get an image of size `size * pixelRatio`. If i pass a RepaintBoundary finder into `matchesGoldenFile` I get an image of size `size`!

Is this the desired result?

**Comment by christian-muertz** [comment]
There seems to be a test for that behaviour:

```dart
group('RenderView', () {
  test('accounts for device pixel ratio in paintBounds', () {
    layout(RenderAspectRatio(aspectRatio: 1.0));
    pumpFrame();
    final Size logicalSize = renderer.renderView.configuration.size;
    final double devicePixelRatio = renderer.renderView.configuration.devicePixelRatio;
    final Size physicalSize = logicalSize * devicePixelRatio;
    expect(renderer.renderView.paintBounds, Offset.zero & physicalSize);
  });
});
```

**Comment by christian-muertz** [comment]
https://github.com/flutter/flutter/pull/17126
https://github.com/flutter/flutter/pull/17129

seem related.

@tvolkert or @Hixie may have some additional knowledge why RenderView.paintBounds account device pixel ratio. I am confused whether `paintBounds` are in logical or physical pixels.

**Comment by tvolkert** [comment]
> Is this the desired result?

No, it's likely an oversight in #17126 that it didn't also apply to `RenderBox`.  @goderbauer do you see any unwanted implications of making `RenderBox.paintBounds` also account for pixel ratio?

**Comment by Hixie** [comment]
RenderView's coordinate system is special (it is the one that gets us from the physical pixels to the logical pixels). That's why it has a special paintBounds.

**Comment by flar** [comment]
RenderView may be performing operations that have to interact with physical pixels, but RenderView itself *is* a RenderObject. Having 2 different definitions of paintBounds depending on the subclass leads to confusion and potential bugs. It could provide a separate devicePaintBounds property for those objects that need to interact with it in device space to avoid confusion.
